### PR TITLE
Fix compilation error with std::min on MacOS

### PR DESCRIPTION
**Pull request by manopapad**
See https://github.com/nv-legate/legate.core/issues/904#issuecomment-1839813308

### CODE REVIEW

File: examples/io/src/read_file.cc [hunk 1]
```diff
@@ -48,7 +48,8 @@ struct read_fn {
     // Compute the absolute offsets to the section that this reader task
     // is supposed to read from the file
     int64_t my_lo = my_id * size / num_readers;
-    int64_t my_hi = std::min((my_id + 1) * size / num_readers, size);
+    int64_t my_hi =
+      std::min(static_cast<int64_t>((my_id + 1) * size / num_readers), static_cast<int64_t>(size));
```
Comment (Jacobfaib) [inline examples/io/src/read_file.cc:52]: I think it is safe to cast just `size` to `int64_t` here. i.e.
```c++
auto ssize = static_cast<int64_t>(size);
int64_t my_hi =
      std::min((my_id + 1) * ssize / num_readers, ssize);
```

Comment (manopapad) [reply examples/io/src/read_file.cc:52]: @CharlelieLrt could you please check if this change works for you also? I can't check myself :-(

Comment (CharlelieLrt) [reply examples/io/src/read_file.cc:52]: Yes, it works.

Comment (manopapad) [reply examples/io/src/read_file.cc]: fe0205d

**Review by Jacobfaib — APPROVED** [review]

**Comment by manopapad** [comment]
CI failures are unrelated, merging

**Pull request merged**